### ISSUE TITLE
Illegal Reflective Access Removal

### ISSUE THREAD

**Issue by MrKorvic** [plan-player-analytics/Plan#1839]
### Is your feature request related to a problem? Please describe.
With the coming updates for Paper to go to Java 11 in the near future, I was prepping to swap over early and found that Plan has an InjectUtil that will likely need to be updated.

```
>.... [15:11:07 INFO]: [Plan] Loading Plan v5.2 build 1174
>.... [15:11:07 INFO]: [Plan] Resolving runtime dependencies..
>.... [15:11:07 INFO]: [Plan] Loading runtime dependencies..
>....WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by net.playeranalytics.plugin.ninja.egg82.utils.InjectUtil (file:/home/container/plugins/Plan-5.2-build-1174.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of net.playeranalytics.plugin.ninja.egg82.utils.InjectUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

### I would like to be able to...
I wasn't quite sure which issue to go with for this since it's working as intended, just not built to be fully Java 11 friendly yet. I couldn't find the source throughout Plan's code at a quick glance, however, I figured I'd share a quick solution that I made for my own local use that Plan may be able to implement without requiring users to update their JVM settings for Plan alone.

https://github.com/Lost-Fables/StartupLoader/blob/main/StartupLoader-common/src/main/java/net/lostfables/startuploadercommon/StartupLoader.java

TL;DR if you create a class that extends `URLClassLoader` and call a super method with the list of URLs you wish to load, Java 11 has no issues with this use and will not throw a fit, and assumably this will continue to be allowed after the illegal access operations are denied in the future.

**Comment by AuroraLS3** [comment]
The InjectUtil is from [egg82/DepDownloader](https://github.com/egg82/DepDownloader), inside Platform-Abstraction-Layer https://github.com/plan-player-analytics/Platform-abstraction-layer/blob/master/api/src/main/java/net/playeranalytics/plugin/dependencies/DependencyLoader.java#L165

DepDownloader is no longer developed, so it'll need to be replaced anyway:
> this library has now been superseded by https://github.com/egg82/MVNLoader which uses the actual Apache Maven parsing libraries rather than hackily attempting to parse relevant information through XML loaders.
> https://github.com/egg82/DepDownloader/pull/1#issuecomment-807915132

AFAIK this warning is only logged by Mohist

**Comment by AuroraLS3** [comment]
The error that occurs with newer JVM
![error log](https://media.discordapp.net/attachments/364111201346781195/830419363596730368/unknown.png)

**Comment by TOWUK** [comment]
use 11-14 java (only linux, on my win10 pc dont work web ui :D)

**Comment by portlek** [comment]
I fix the error with this;
```java
/**
 * disable some of warning with a tricky way.
 */
private static void disableWarning() {
  try {
    final var theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
    theUnsafe.setAccessible(true);
    final var u = (Unsafe) theUnsafe.get(null);
    final var cls = Class.forName("jdk.internal.module.IllegalAccessLogger");
    final var logger = cls.getDeclaredField("logger");
    u.putObjectVolatile(cls, u.staticFieldOffset(logger), null);
  } catch (final Exception e) {
    // ignore
  }
}
```

**Comment by AuroraLS3** [comment]
> I fix the error with this
Hmm, that seems like ignoring the warning instead of fixing the issue 🤔 

The cause needs to be addressed because as seen above the plugin will break with newer Java versions (I assume 16+) if not addressed.

**Comment by portlek** [comment]
> > I fix the error with this
> > Hmm, that seems like ignoring the warning instead of fixing the issue thinking
> 
> The cause needs to be addressed because as seen above the plugin will break with newer Java versions (I assume 16+) if not addressed.

yea this should be fixed in the dependency.
